kukur/source/delta/delta_lake.py

```diff
@@ -45,25 +141,206 @@ def open_child(self, subpath: str):
         raise NotImplementedError()
 
 
-class DeltaLakeSource(BaseArrowSource):
+class DeltaLakeSource:
     """Connect to a Delta Lake."""
 
-    def read_file(self, file_like) -> Table:
+    __loader: Loader
+    __options: DeltaSourceOptions
+    __quality_mapper: QualityMapper
+    __sort_by_timestamp: bool = False
+
+    def __init__(
+        self,
+        options: DeltaSourceOptions,
+        loader: Loader,
+        quality_mapper: QualityMapper,
+        *,
+        sort_by_timestamp: bool = False,
+    ):
+        """Create a new data source."""
+        self.__loader = loader
+        self.__options = options
+        self.__quality_mapper = quality_mapper
+        self.__sort_by_timestamp = sort_by_timestamp
+
+    def read_file(self, file_like) -> pa.Table:
         """Return a PyArrow Table for the Delta Table at the given URI."""
         return DeltaTable(file_like).to_pyarrow_table()
 
+    def _format_tag_partition(
+        self,
+        partition: DeltaLakePartition,
+        selector: SeriesSelector,
+    ) -> Tuple[str, str, str]:
+        column_name = self.__options.column_mapping.get(partition.key, partition.key)
+        return (column_name, "=", selector.tags[partition.key])
+
+    def read_partitioned_file(
```

Comment:
This also shouldn't be public.

We should have `search`, `get_metadata` and `get_data` on top of the file.